docusaurus/docs/ui-extensions/register-an-extension.md

```diff
@@ -0,0 +1,128 @@
+   - `category` - a category that we should use to group your extension with other extensions.
+   - `icon` - an icon that should be used while displaying your extension.
+   - `configure` - a function that will be called prior to displaying the extension which enables you to dynamically change/hide your extension depending on the context.
+
+::: note
+
+One of `path` or `onClick` is required. Otherwise the extension will be hidden due to incorrect configuration.
+
+:::
+
+### Example: Adding a regular link
+
+In the example below we are adding an extension link to the Grafana dashboard panel menu. When the user clicks the menu item "Go to basic app" he/she will be navigated to `/a/myorg-basic-app/one`.
+
+```typescript
+new AppPlugin().configureExtensionLink({
+  title: 'Go to basic app',
+  description: 'Will send the user to the basic app',
+  extensionPointId: 'grafana/dashboard/panel/menu',
+  path: '/a/myorg-basic-app/one', // Must start with "/a/<PLUGIN_ID>/"
+});
+```
+
+### Example: Adding a regular link with query string parameters from the context
+
+In the example below we are adding an extension link to the Grafana dashboard panel menu. When the user click the menu item "Go to basic app" he/she will be navigated to `/a/myorg-basic-app/one?panelId=12345&timeZone=utc`.
+
+```typescript
+new AppPlugin().configureExtensionLink({
+  title: 'Go to basic app',
+  description: 'Will send the user to the basic app',
+  extensionPointId: 'grafana/dashboard/panel/menu',
+  path: '/a/myorg-basic-app/one', // Must start with "/a/<PLUGIN_ID>/"
+  configure: (context: PanelContext) => {
+    const {timeZone, panelId } = context;
+
+    // You only need to return the properties that you would like to override.
+    return {
+      path: `/a/myorg-basic-app/one?panelId=${panelId}&timeZone=${timeZone}`
+    };
+  }
+});
+```
+
+### Example: Adding a regular link that should be visible depending on the context
+
+In the example below we are adding an extension link to the Grafana dashboard panel menu. It will only be visible for panels with time zone set to UTC.
+
+```typescript
+new AppPlugin().configureExtensionLink({
+  title: 'Go to basic app',
+  description: 'Will send the user to the basic app',
+  extensionPointId: 'grafana/dashboard/panel/menu',
+  path: '/a/myorg-basic-app/one', // Must start with "/a/<PLUGIN_ID>/"
+  configure: (context: PanelContext) => {
+    const {timeZone } = context;
+
+    switch (toLowerCase(timeZone)) {
+      case 'utc':
+        return {}; // no overrides applied but we want to display the extension.
+      default:
+        return undefined; // returning undefined from the configure function will hide the extension.
+    }
+  }
+});
+```
+
+### Example: Adding a link that should display a modal
+
+In the example below we are adding an extension link to the Grafana dashboard panel menu. It will open a flow (defined in our app) in a modal on top of the current view.
```

Comment:
```suggestion
In the following example, we add an extension link to the Grafana dashboard panel menu. It will open a flow (defined in our app) in a modal on top of the current view.
```